app/forms/funding/training_initiatives_form.rb

```diff
@@ -21,5 +22,9 @@ def compute_fields
     def form_store_key
       :training_initiative
     end
+
+    def training_initiative_required?
+      trainee.record_source != RecordSources::API
```

Comment:
Will this cause issue if we firstly added via api then edit via ui?

Reply:
yeah quite possibly...should we always set record_source to API in any update actions?

Reply:
so when updating errr...

- We can only update records that are `api` vs others
- We set any records source to be `api` regardless

Feels like we may need to have a little thing